pkg/synopsysctl/cmd_update.go

```diff
@@ -1322,6 +1322,11 @@ var updatePolarisCmd = &cobra.Command{
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
+		helmRelease, err := util.GetWithHelm3(polarisName, namespace, kubeConfigPath)
+		if err != nil {
+			return fmt.Errorf("failed to get previous user defined values: %+v", err)
+		}
+		updatePolarisCobraHelper.SetArgs(helmRelease.Config)
```

Comment:
helmRelease.Config will only be the values set by `--set` and `-f`. If you want all the values in the Release then you need to use `helmRelease.Chart.Values`

Reply:
not required all the values @mphammer . We need only the user specific flags bcoz the new chart values need to be applied for the default settings

Reply:
What happens if the default port for the old chart is 1000 and the default port for the new chart is 1234. And the user wants to use port 1000 but they didn't specify it with flags. Then they will be changed to use port 1234.

Reply:
@mphammer I would say that is expected behavior and it's also just the problem with the helm chart.  Another point that makes it very clear that the data and data structure in values.yaml is super critical and we should be careful about what we put there and not change it often.